src/main/java/com/github/manolo8/darkbot/Bot.java

```diff
@@ -32,10 +41,86 @@ public static void main(String[] args) throws IOException {
+                        long externalStartTime = 0;
+                        if (externalProcessHandle != null) {
+                            ProcessHandle.Info ExternalProcessInfo = externalProcessHandle.info();
+                            externalStartTime = ExternalProcessInfo.startInstant().map(Instant::toEpochMilli).orElse(0L);
+                        }
+
+                        if (ProcessHandle.of(filePid).isPresent() && externalStartTime == fileStartTime) {
+                            Files.writeString(filePath, "\n" + currentPid + " " + currentStartTime, StandardOpenOption.APPEND);
+
+                            JButton proceed = new JButton("Proceed Anyways", UIUtils.getIcon("diagnostics"));
+                            JButton cancel = new JButton("Cancel");
+                            AtomicInteger result = new AtomicInteger(-1);
+
+                            proceed.addActionListener(a -> {
+                                SwingUtilities.getWindowAncestor(proceed).setVisible(false);
+                                result.set(0);
+                            });
+                            cancel.addActionListener(a -> {
+                                SwingUtilities.getWindowAncestor(cancel).setVisible(false);
+                                result.set(1);
+                            });
+
+                            Popups.of("Multiple bots on the same folder",
+                                            "You're currently running multiple bot instances from the same folder.\n" +
+                                            "This can cause crash, unexpected problems, and some broken functionality.\n" +
+                                            "Please create a separate folder to run multiple instances.",
+                                            JOptionPane.WARNING_MESSAGE)
+                                    .options(new Object[]{proceed, cancel})
+                                    .initialValue(cancel)
+                                    .showOptionSync();
+
+                            if (result.get() == 1) {
+                                System.out.println("Reject multiple bot warning, closing bot");
+                                System.exit(100);
+                            }
+                            break;
+                        } else {
+                            if (i == fileContent.size() - 1)
+                                Files.writeString(filePath, currentPid + " " + currentStartTime, StandardOpenOption.TRUNCATE_EXISTING);
+                        }
+                    } catch (java.lang.NumberFormatException e) {
+                        e.printStackTrace();
+                        Files.writeString(filePath, currentPid + " " + currentStartTime, StandardOpenOption.TRUNCATE_EXISTING);
+                    }
+                }
+            }
+        } else {
+            Files.writeString(filePath, currentPid + " " + currentStartTime, StandardOpenOption.CREATE);
+        }
```

Comment:
I would again insist into not having this write string 3 times, instead having it as a final step outside all the ifs